### ISSUE TITLE
Feature Request: auto detect author

### ISSUE THREAD

**Issue by shoichikaji** [tokuhirom/OrePAN2#34]
Currently when injecting, I have to specify author, otherwise it defaults to DUMMY.

It would be nice OrePAN2::Injector has a option to detect author automatically if possible.
I think x_authority in META.json is a good start point.

**Comment by shoichikaji** [comment]
Or, author accepts code reference :-)
```perl
use CPAN::Meta;
use OrePAN2::Injector;

my $detect_author = sub {
    my $author = "DUMMY";
    my $file = "META.json";
    if (-f $file) {
        my $meta = CPAN::Meta->load_file($file);
        if (my $authority = $meta->{x_authority}) {
            $authority =~ s/^cpan://i;
            $author = uc $authority;
        }
    }
    $author;
};

my $injector = OrePAN2::Injector->new(directory => "repo");

$injector->inject("git://github.com/tokuhirom/OrePAN2.git", {
    author => $detect_author,
});
```

**Comment by oalders** [comment]
I like the idea, but x_authority is not always going to be accurate.

https://metacpan.org/source/OALDERS/OrePAN2-0.36/META.json#L124

That would put the author tarball under TOKUHIROM but on CPAN it would be under OALDERS, which would be confusing.  :)

In the case of the git repo, it probably doesn't matter so much.  In the case of a CPAN mirror it's easy enough to get the author name from the URL.  The coderef is an interesting idea, but we'd really need to think about the args which it would require, since there might be more than one way to try to get an author name (based on URL, contents of tarball, etc)

@tokuhirom what do you think?

**Comment by shoichikaji** [comment]
@oalders Thank you for your comment!
Yes, you're right, x_authority is not always accurate, and it seems that there is no way to detect author accurately.

So my request is that `author` parameter accepts a code reference with
* the arguments of it are `my ($source) = @_` (here `$source` is a argument of inject() method)
* the working directory of it is the top of the distribution (then we need to extract tarball before calling the code reference)

**Comment by oalders** [comment]
@shoichikaji This sounds reasonable to me.  Are you willing to provide the patch?

**Comment by shoichikaji** [comment]
implemented by #35
